### ISSUE TITLE
[PATCH] Use default values for sample rate, frames/period, and periods/buffer

### ISSUE THREAD

**Issue by kmatheussen** [rncbc/qjackctl#112]
Hi,

It seems like jack can fail on Windows because it sets options for "-r", and "-p". Here is a patch (and two new files) to fix this:
http://users.notam02.no/~kjetism/qjackctl_defaults.diff
http://users.notam02.no/~kjetism/qjackctlPeriodsSpinBox.cpp
http://users.notam02.no/~kjetism/qjackctlPeriodsSpinBox.h

Sorry for not making a pull request. (My qjackctl is set up to compile for windows, and I don't want to screw up things.)

It works by first adding "(default)" options to "Sample Rate", "Frames/period", and "Periods/buffer", and then let the command line generator ignore these variables if set to "(default)"

**Comment by kmatheussen** [comment]
Here's the bug report: http://users.notam02.no/~kjetism/radium/forum/viewtopic.php?f=7&p=906&sid=b32c12dbd81b382cfd4c3b41c14b55a9#p906

**Comment by kmatheussen** [comment]
(I've uploaded new .diff file. The old one had a lot of unnecessary things in the .ui diff added by qt-designer)

**Comment by rncbc** [comment]
patches are applied to new branch (kmatheussen_112)[https://github.com/rncbc/qjackctl/tree/kmatheussen_112]

thanks

**Comment by kmatheussen** [comment]
Thank you. I think this is an important patch that should be applied.I don't think it breakes anything.

**Comment by rncbc** [comment]
patch have been slightly refactored on new branch [kmatheussen_112](https://github.com/rncbc/qjackctl/tree/kmatheussen_112)...

soon, some other parameters will follow in the same vein...
thanks

**Comment by rncbc** [comment]
just to let you know this has been merged to [develop](https://github.com/rncbc/qjackctl/tree/develop)
please refer to the latter as branch kmatheussen_112 shall be deleted soon in the new year ;)

**Comment by kmatheussen** [comment]
Looks good, great work!

**Comment by falkTX** [comment]
Is this part of the 0.9.0 release?

**Comment by rncbc** [comment]
> Is this part of the 0.9.0 release?

no. it's in [develop](https://github.com/rncbc/qjackctl/tree/develop) branch.

**Comment by falkTX** [comment]
This is part of 626731e752b4694051b3d447cebb90116fd37186 then I assume..?

**Comment by rncbc** [comment]
> This is part of [626731e](https://github.com/rncbc/qjackctl/commit/626731e752b4694051b3d447cebb90116fd37186) then I assume..?

yes

**Comment by falkTX** [comment]
Cool, I will be doing some testing with that commit applied then.

It is quite the important change. I plan to do a new jack2 release on the 15th of this month, would you mind doing a qjackctl release before that? Otherwise I just apply the change on top of 0.9.0, it is just a little bit misleading in terms of versioning.

Thanks a lot!

**Comment by rncbc** [comment]
until then, you can always checkout from [develop](https://github.com/rncbc/qjackctl/tree/develop) branch.
